src/data_tools/include/data_tools/jsf_data.h

```diff
@@ -43,13 +46,13 @@ struct jsf_sss_ping
     long long time_stamp_; // posix time stamp
     jsf_sss_ping_side port;
     jsf_sss_ping_side stbd;
-    bool is_bad;
+    int channel_num; // channel number, 0 -> port, 1 -> starboard
 
 
 	template <class Archive>
     void serialize( Archive & ar )
     {
-        ar(CEREAL_NVP(first_in_file_), CEREAL_NVP(time_string_), CEREAL_NVP(time_stamp_), CEREAL_NVP(port), CEREAL_NVP(stbd), CEREAL_NVP(is_bad));
+        ar(CEREAL_NVP(first_in_file_), CEREAL_NVP(time_string_), CEREAL_NVP(time_stamp_), CEREAL_NVP(port), CEREAL_NVP(stbd), CEREAL_NVP(channel_num));
```

Comment:
Did we not need the bad flag anymore?

Reply:
The bad flag is counted as "if two pings has the same time stamp but they have the same channel number", which is now going be detected inside of jsf_sss_ping::PingsT parse_file<jsf_sss_ping>(const boost::filesystem::path& file). If there is a bad ping (usually in the end of file), there will be printing "data format is invalid ..." and the channel numbers respectively as well.